frame/utility/src/tests.rs

```diff
@@ -638,3 +668,92 @@ fn force_batch_works() {
 		System::assert_last_event(utility::Event::BatchCompletedWithErrors.into());
 	});
 }
+
+#[test]
+fn submit_call_on_behalf_of_works() {
+	new_test_ext().execute_with(|| {
+		let signer = crypto::create_ed25519_pubkey(b"//verifier".to_vec());
+		let signer_id = signer_to_account_id(&signer.clone());
+		let call = Box::new(Call::System(SystemCall::remark { remark: vec![] }));
+
+		use frame_system::{Account, AccountInfo};
+		Account::<Test>::insert(signer_id, AccountInfo::default());
+
+		let forwarded_call = ForwardedCall {
+			nonce: <Test as frame_system::Config>::Index::zero(),
+			deadline: 1,
+			call: call.clone(),
+			signer: signer.clone(),
+			whitelisted_forwarder: None,
+		};
+		let signature =
+			crypto::create_ed25519_signature(&Encode::encode(&forwarded_call), signer.clone());
+
+		assert_ok!(Utility::submit_call_on_behalf_of(
+			Origin::signed(1),
+			forwarded_call.clone(),
+			signature.clone()
+		));
+
+		// validate event
+		System::assert_last_event(
+			utility::Event::CallForwarded {
+				forwarder: 1,
+				new_origin: signer_id,
+				call_hash: blake2_256(&Encode::encode(&call)),
```

Comment:
how does this work? how do you successfully sign a message with account "1"?

Seems you need to use real accountid32s here

Reply:
`forwarder` has the AccountId type, it's a regular account that will submit the tx and pay the fee